expression/builtin.go

```diff
@@ -226,6 +239,17 @@ func (b *baseBuiltinFunc) evalDuration(row []types.Datum) (types.Duration, bool,
 	return val.GetMysqlDuration(), false, errors.Trace(err)
 }
 
+func (b *baseBuiltinFunc) evalJSON(row []types.Datum) (json.JSON, bool, error) {
```

Comment:
I prefer return `JSON` instead of `*JSON`, because I think the struct is not too large, and it avoids cache misses or heap allocations.